java/mtsj/core/src/main/java/io/oasp/application/mtsj/predictionmanagement/dataaccess/impl/dao/PredictionDayDataDaoImpl.java

```diff
@@ -0,0 +1,199 @@
+    getEntityManager().createNativeQuery("DELETE FROM TMP_PREDICTION_FIT").executeUpdate();
+    getEntityManager().createNativeQuery(
+      "CALL _SYS_AFL.PAL_AUTOARIMA(TMP_PREDICTION_DATA, PREDICTION_AUTOARIMA_PARAMS, "
+      + "TMP_PREDICTION_MODEL, TMP_PREDICTION_FIT) WITH OVERVIEW"
+    ).executeUpdate();
+
+    // Save model
+
+    CriteriaDelete<PredictionModelDataEntity> criteriaDelete = cb.createCriteriaDelete(PredictionModelDataEntity.class);
+    Root<PredictionModelDataEntity> criteriaDeleteRoot = criteriaDelete.from(PredictionModelDataEntity.class);
+    criteriaDelete.where(cb.equal(criteriaDeleteRoot.get("dish").get("id"), dish.getId()));
+    getEntityManager().createQuery(criteriaDelete).executeUpdate();
+
+    getEntityManager().createNativeQuery(
+      "INSERT INTO PREDICTION_ALL_MODELS SELECT ?1 AS idDish, key, value "
+      + "FROM TMP_PREDICTION_MODEL"
+    )
+    .setParameter(1, dish.getId())
+    .executeUpdate();
+
+    PredictionModelDataEntity entity = new PredictionModelDataEntity();
+    entity.setDish(dish);
+    entity.setKey("_date");
+    entity.setValue(startDate.toString());
+    getEntityManager().persist(entity);
+  }
+
+  public void generatePredictionFor(DishEntity dish) {
+
+    // Prepare model parameters
+
+    getEntityManager().createNativeQuery("DELETE FROM TMP_PREDICTION_MODEL").executeUpdate();
+    getEntityManager().createNativeQuery(
+      "INSERT INTO TMP_PREDICTION_MODEL SELECT "
+      + "key, value FROM PREDICTION_ALL_MODELS "
+      + "WHERE idDish = ?1"
+    )
+    .setParameter(1, dish.getId())
+    .executeUpdate();
+
+    // Do prediction
+
+    getEntityManager().createNativeQuery("DELETE FROM TMP_PREDICTION_FORECAST").executeUpdate();
+    getEntityManager().createNativeQuery(
+      "CALL _SYS_AFL.PAL_ARIMA_FORECAST(PREDICTION_FORECAST_DATA, TMP_PREDICTION_MODEL, "
+      + "PREDICTION_ARIMA_PARAMS, TMP_PREDICTION_FORECAST) WITH OVERVIEW"
+    ).executeUpdate();
+
+    // Save prediction
+
+    CriteriaBuilder cb = getEntityManager().getCriteriaBuilder();
+    CriteriaDelete<PredictionDayDataEntity> criteriaDelete = cb.createCriteriaDelete(PredictionDayDataEntity.class);
+    Root<PredictionDayDataEntity> from = criteriaDelete.from(PredictionDayDataEntity.class);
+    criteriaDelete.where(cb.equal(from.get("dish").get("id"), dish.getId()));
+    getEntityManager().createQuery(criteriaDelete).executeUpdate();
+
+    getEntityManager().createNativeQuery(
+      "INSERT INTO PREDICTION_ALL_FORECAST (idDish, dishName, timestamp, forecast) "
+      + "SELECT ?1 AS idDish, Dish.name AS dishName, timestamp, forecast "
+      + "FROM TMP_PREDICTION_FORECAST LEFT JOIN Dish ON Dish.id = ?1"
+    )
+    .setParameter(1, dish.getId())
+    .executeUpdate();
+  }
+
+  @Override
+  public List<PredictionDayDataEntity> getPrediction(PredictionCriteriaEto criteria) {
+
+    List<DishEntity> dishes = getDishDao().findDishs(new DishSearchCriteriaTo()).getResult();
```

Comment:
If I am not mistaken, you are querying for all dishes here, correct? As Dishes are master data this is feasible but then we should just use `findAll()`.

Reply:
`findAll()` isn't available without casting, There is only `findAll(Iterable<Long> ids)`.